src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -530,6 +535,25 @@ void MergeTreeData::checkProperties(
 
         for (const auto & index : new_metadata.secondary_indices)
         {
+            if (!allow_suspicious_indices && !attach)
+            {
+                NameSet index_key_set;
+                const auto * index_ast = index.definition_ast->as<ASTIndexDeclaration>();
+
+                if (const auto * tuple_ast = index_ast->expr->as<ASTFunction>())
+                {
+                    for (const auto & child : tuple_ast->arguments->children)
+                    {
+                        const auto &tree_hash = child->getTreeHash();
+                        const String key = toString(tree_hash.first) + "_" +toString(tree_hash.second);
```

Comment:
```suggestion
                        const String key = toString(tree_hash.first) + "_" + toString(tree_hash.second);
```
try using `fmt::format(...)`

Reply:
Perhaps the first part of the hash only will be sufficient to identify duplicates.

Reply:
done